pkg/k8s/allowlist.go

```diff
@@ -16,231 +16,247 @@ package k8s
+		"IAMServiceAccountKey",
+		"IAMWorkforcePool",
+		"IAMWorkforcePoolProvider",
+		"IAMWorkloadIdentityPool",
+		"IAMWorkloadIdentityPoolProvider",
+		"IAPBrand",
+		"IAPIdentityAwareProxyClient",
+		"IdentityPlatformConfig",
+		"IdentityPlatformOAuthIDPConfig",
+		"IdentityPlatformTenant",
+		"IdentityPlatformTenantOAuthIDPConfig",
+		"KMSCryptoKey",
+		"KMSKeyRing",
+		"LoggingLogBucket",
+		"LoggingLogExclusion",
+		"LoggingLogMetric",
+		"LoggingLogSink",
+		"LoggingLogView",
+		"MemcacheInstance",
+		"MonitoringAlertPolicy",
+		"MonitoringDashboard",
+		"MonitoringGroup",
+		"MonitoringMetricDescriptor",
+		"MonitoringMonitoredProject",
+		"MonitoringNotificationChannel",
+		"MonitoringService",
+		"MonitoringServiceLevelObjective",
+		"MonitoringUptimeCheckConfig",
+		"NetworkConnectivityHub",
+		"NetworkConnectivitySpoke",
+		"NetworkSecurityAuthorizationPolicy",
+		"NetworkSecurityClientTLSPolicy",
+		"NetworkSecurityServerTLSPolicy",
+		"NetworkServicesEndpointPolicy",
+		"NetworkServicesGRPCRoute",
+		"NetworkServicesGateway",
+		"NetworkServicesHTTPRoute",
+		"NetworkServicesMesh",
+		"NetworkServicesTCPRoute",
+		"NetworkServicesTLSRoute",
+		"OSConfigGuestPolicy",
+		"OSConfigOSPolicyAssignment",
+		"PrivateCACAPool",
+		"PrivateCACertificate",
+		"PrivateCACertificateAuthority",
+		"PrivateCACertificateTemplate",
+		"Project",
+		"PubSubLiteReservation",
+		"PubSubSchema",
+		"PubSubSubscription",
+		"PubSubTopic",
+		"RecaptchaEnterpriseKey",
+		"RedisInstance",
+		"ResourceManagerLien",
+		"ResourceManagerPolicy",
+		"RunJob",
+		"RunService",
+		"SQLDatabase",
+		"SQLInstance",
+		"SQLSSLCert",
+		"SQLUser",
+		"SecretManagerSecret",
+		"SecretManagerSecretVersion",
+		"Service",
+		"ServiceDirectoryEndpoint",
+		"ServiceDirectoryNamespace",
+		"ServiceDirectoryService",
+		"ServiceIdentity",
+		"ServiceNetworkingConnection",
+		"SourceRepoRepository",
+		"SpannerDatabase",
+		"SpannerInstance",
+		"StorageBucket",
+		"StorageBucketAccessControl",
+		"StorageDefaultObjectAccessControl",
+		"StorageNotification",
+		"StorageTransferJob",
+		"TagsTagBinding",
+		"TagsTagKey",
+		"TagsTagValue",
+		"VPCAccessConnector",
+	}
+
+	// v1beta1KindsWithComputedFieldsUnderStatus contains all the existing
+	// v1beta1 kinds that have computed fields directly under 'status' in the
+	// schema. This list includes all the kinds in
+	// v1beta1KindsWithStateIntoSpecMergeSupport and
+	// 'ComputeNetworkFirewallPolicyAssociation'.
+	// Any newly supported v1beta1 kinds should NOT have computed fields
+	// directly under 'status' in the schema.
+	v1beta1KindsWithComputedFieldsUnderStatus = append(v1beta1KindsWithStateIntoSpecMergeSupport,
+		"ComputeNetworkFirewallPolicyAssociation")
+)
 
-func SupportsStateIntoSpecMergeInKind(kind string) bool {
-	return isValueInAllowlist(kind, v1beta1KindsWithStateIntoSpecMergeSupport)
+func SupportsStateIntoSpecMerge(gvk schema.GroupVersionKind) bool {
+	if gvk.Version != KCCAPIVersionV1Beta1 {
```

Comment:
With this logic change (from #1438 ), we are going to break all the v1alpha1 resources: All the new v1alpha1 resource (KRM object) will only support "state-into-spec: absent".

Looks to be an okay change to me. @justinsb as you've showed concern about breaking v1alpha1 resources, I just want to confirm you are aware of this change.